wallet/src/de/schildbach/wallet/service/platform/PlatformSyncService.kt

```
@@ -913,19 +924,23 @@ class PlatformSynchronizationService @Inject constructor(
                             log.info("processing TxMetadata: barcode format change")
                             if (cachedItems.find {
-                                    it.txId == txIdAsHash && it.cacheTimestamp > doc.createdAt!! &&
+                                    it.txId == txIdAsHash && it.cacheTimestamp > doc.updatedAt!! &&
                                         it.barcodeFormat != null && it.barcodeFormat != barcodeFormat
                                 } == null
                             ) {
                                 log.info("processing TxMetadata: barcode value change: changing value")
-                                giftCard.barcodeFormat = BarcodeFormat.valueOf(barcodeFormat)
+                                try {
+                                    giftCard.barcodeFormat = BarcodeFormat.valueOf(barcodeFormat)
+                                } catch (e: IllegalArgumentException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)